### ISSUE TITLE
document the hs_url parameter

### ISSUE THREAD

**Issue by uhoreg** [vector-im/riot-web#5469]
which was added in matrix-org/matrix-react-sdk#716 but doesn't seem to be documented anywhere

**Comment by uhoreg** [comment]
to clarify, this is about being able to do things like `https://riot.im/#/register?hs_url=matrix.whatever, and not about the `config.json` parameters

**Comment by dagwieers** [comment]
Why not contribute a PR with exactly what you expect ?

**Comment by t3chguy** [comment]
Yes https://github.com/vector-im/riot-web/issues/9908

**Comment by aaronraimist** [comment]
@t3chguy this one had more thumbs so maybe this should turn into the fix hs_url issue

**Comment by t3chguy** [comment]
Sure.

**Comment by t3chguy** [comment]
Taking over https://github.com/vector-im/riot-web/issues/9908

**Comment by okias** [comment]
Would be really great to have it fixed.

**Comment by danger89** [comment]
I can't agree more, especially if matrix want to become more decentralized and more federated.

**Comment by bwindels** [comment]
I wonder if phishing is a concern here as well,
getting people to try and log in on a malicious homeserver

**Comment by binarymaster** [comment]
Out of curiosity I have tried to change the link to this:

https://app.element.io/#/login&hs_url=https://matrix.melroy.org (notice the hs part goes after the hash, not part of URL query)

And for some reason it makes Element hang on loading, no matter what homeserver is specified.

**Comment by timmc** [comment]
Yeah, I've found that the lack of this is a serious usability problem for the people I've invited to chat on my server. After they register they go to element.io and then they have to do a complicated little dance and pay attention to terminology that's new to them ("homeserver") in order to sign in properly.

**Comment by PxBCPG** [comment]
how is it going?

**Comment by Babalion** [comment]
Hey guys,

in my opinion it shouldn't be that difficult to plug in a string from an url-parameter...
It would be nice for the initial user experience to have this feature working!

Maybe I am able to create a pull request but i am completely new to the element development.
Where exactly is this thing handled? Which files does one have to change?

Thanks!

**Comment by benjaoming** [comment]
@Babalion that sounds great, but a suggestion is already up in a PR, you could help out there? https://github.com/vector-im/element-web/pull/17824

**Comment by alexander-potemkin** [comment]
It's a thing working on iOS, guess it makes perfect sense from the user perspective to have things working unified across the platforms.

**Comment by Orangestar12** [comment]
It's been a while without any word. Is this option ever coming back?
